### ISSUE TITLE
[1.15+] `timeout` in a batch script instantly exits a defterm window

### ISSUE THREAD

**Issue by zadjii-msft** [microsoft/terminal#14499]
Oh god this is so much worse in 1.15. It doesn't timeout at all, it instantly exits!

```cmd
echo foo
@rem set /p id=Enter ID: 
timeout /t 10
echo bar
exit 1234
```

![image](https://user-images.githubusercontent.com/18356694/205757385-aa1719f2-cffd-41ee-8d9b-a8ec19051610.png)

_Originally posted by @zadjii-msft in https://github.com/microsoft/terminal/issues/12951#issuecomment-1338357561_

**Comment by zadjii-msft** [comment]
~OH oh oh I bet this has to do with focus events. That's my guess - `timeout` will bail on input. Like, any input? That I'm not sure about. But I bet the regression between 1.12 when this was filed, and now (1.15/16/17) is the focus event stuff, and I bet eveything is immediately getting a focus event in the input queue right off the bat.~ 

I'll need to go inspect `timeout`'s source to confirm. That's my hypothesis. 


Notably, this doesn't repro the same if you actually "restart" the batch script, until you press a key: 
![image](https://user-images.githubusercontent.com/18356694/205942092-7f79e221-790c-47cc-8ee5-9df8cf558c8a.png)

I noticed this when I restored the Terminal from being quit, and this defterm window came back ~(presumably, without the focus event)~. This might be related to #13514 


Okay, my hypothesis was wrong. Looks like `timeout` does filter FOCUS events out... even if it does have some other horrifying code in it.
